### PR TITLE
Fixing issue with attaching Python dictionary to Link objects. For so…

### PR DESCRIPTION
**Pull request by maciekswat**
…me reason #ifdef SWIGPYTHON macro did not work with swig 4.0. I moved dictionary creation to CompuCellExtraDeclarations.i.

### CODE REVIEW

**Comment by tjsego** [comment]
I can't reproduce this with SWIG 4.1.1. Can you try rebuilding with this version and see if this fix is necessary before proceeding?

**Comment by maciekswat** [comment]
I can try it but I would rather keep functions like this inside the swig file, because it is e to easier to understand the code. It took me a while to figure out the logic of it because I did not expect to find code to find it in C++ file. I think grouping python-wrapper related code inside python wrapper generating code is easier to maintain

**Review by tjsego — APPROVED** [review]
Looks good

File: CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinks.h [hunk 1]
```diff
@@ -193,9 +193,14 @@ namespace CompuCell3D {
...
-            if (pyAttrib == nullptr) pyAttrib = PyDict_New();
-#endif
+//#ifdef SWIG
+//            if (pyAttrib == nullptr){
+//                PyGILState_STATE gstate;
+//                gstate = PyGILState_Ensure();
+//                pyAttrib = PyDict_New();
+//                PyGILState_Release(gstate);
+//            }
+//#endif
```
[the remark this answers is not in the document]
Comment (maciekswat) [reply CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinks.h:203]: yep, I wanted to leave it as a footprint ;)But we have this already in our history

[the remark this answers is not in the document]
Comment (maciekswat) [reply CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinks.h]: Done

**Pull request merged**